engine/src/main/java/io/zeebe/engine/state/DefaultZeebeDbFactory.java

```diff
@@ -11,34 +11,68 @@
 import io.zeebe.db.ZeebeDbFactory;
 import io.zeebe.db.impl.rocksdb.ZeebeRocksDBMetricExporter;
 import io.zeebe.db.impl.rocksdb.ZeebeRocksDbFactory;
+import java.util.Properties;
 import java.util.function.BiFunction;
 
 public final class DefaultZeebeDbFactory {
 
+  public static final BiFunction<String, ZeebeDb<ZbColumnFamilies>, ZeebeRocksDBMetricExporter>
+      DEFAULT_DB_METRIC_EXPORTER_FACTORY =
+          (partitionId, database) ->
+              new ZeebeRocksDBMetricExporter<>(partitionId, database, ZbColumnFamilies.class);
+  private static ZeebeDbFactory<ZbColumnFamilies> instance;
+
   /**
-   * The default zeebe database factory, which is used in most of the places except for the
+   * Returns the default zeebe database factory, which is used in most of the places except for the
    * exporters.
+   *
+   * @return the created zeebe database factory
    */
-  public static final ZeebeDbFactory<ZbColumnFamilies> DEFAULT_DB_FACTORY =
-      defaultFactory(ZbColumnFamilies.class);
+  public static ZeebeDbFactory<ZbColumnFamilies> defaultFactory() {
```

Comment:
I had to replace this global constant with a method to allow different ways of constructing the factory (for different tests). It is now replaced by a Singleton pattern